Sound ingestion

liermannj opened the pull request:
Première partie : Ingestion signal sonore + découpage en morceaux cohérents correspondants à des ensembles de sons "ininterrompus" (ou très peu interrompus).

Algo : 

Ingestion depuis micro => conversion en onde sonore (type vaguelette) => séparation en fenêtres d'intervalles de temps fixes (avec overlapping) => hamming des fenêtres (je sais pas le traduire...) => détection des fenêtres parlées et silence => groupement des fenêtres parlées consécutives

tl;dr : Source de morceaux de sons correspondants à un mot

Commit:
feat - utils: add utils module

- to be reused with other modules
- tried to be as generic as possible
- add graphs and test utils

Commit:
feat - sound: add sound module

- module does sound ingestion
- output is coherent frames of words
- add sample boot for quickstart

01101101 commented on sound/src/main/scala/com/jliermann/sound/output/OutputLive.scala:
```diff
@@ -0,0 +1,20 @@
+    StreamConverters
+      .fromOutputStream(() => new FileOutputStream(file))
+      .contramap[String](ByteString(_))
+      .contramap[Any](any => s"$any\n")
```
`Any` 😀 
```suggestion
    Flow[Any]
      .map(any => s"$any\n")
      .map(ByteString(_))
      .to(FileIO.toPath(file.toPath))
```

01101101 replied:
Je trouve ça plus facile de raisonner "dans le sens" du stream.
`FileIO` vient de la : https://doc.akka.io/docs/akka/current/stream/stream-io.html#streaming-file-io

liermannj replied:
Opté pour toMat pour garder le même type de sortie :

liermannj replied:
on perd le type dans ce sens là, au final j'ai fait toMat + Keep.right, mais j'ai pas l'impression que ce soit super utile, la doc a l'air de déprécater IOResult

Commit:
improve - all: move OutputLive to utils + add test for it

- correct implementation to match PR change too

Commit:
improve - utils: add a bit more genericity

Commit:
improve - sound: extract raw reading

- did this to add more tests to audio input
- added an it with a file saying "bonjour la souris"
- noticed inconsistency in the results, should investigate

Commit:
fix - sound: manage targetDataLine correctly
